web/src/lib/components/shared-components/search-bar/search-filter-box.svelte

```diff
@@ -282,11 +231,13 @@
 
       filter = {
         context: 'query' in searchQuery ? searchQuery.query : '',
-        people: orderBySelectedPeopleFirst(personIds.map((id) => ({ id }))),
+        people: orderBySelectedPeopleFirst(
+          personIds.map((id) => suggestions.people.find((person) => person.id === id)!),
+        ),
```

Comment:
I mentioned this in the [original PR](https://github.com/immich-app/immich/pull/6968#discussion_r1484959291) as well, but `filter.people`  doesn't need the person object, just the IDs. This scales quadratically for no reason.

Reply:
Changed to using `Set<string>` with person ID